url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/semantics/examples/TreeBuilderExample.java

```diff
@@ -0,0 +1,37 @@
+package fri.patterns.interpreter.parsergenerator.semantics.examples;
+
+import java.io.*;
+import fri.patterns.interpreter.parsergenerator.*;
+import fri.patterns.interpreter.parsergenerator.lexer.*;
+import fri.patterns.interpreter.parsergenerator.syntax.Syntax;
+import fri.patterns.interpreter.parsergenerator.syntax.builder.*;
+import fri.patterns.interpreter.parsergenerator.semantics.TreeBuilderSemantic;
+
+/**
+	TreeBuilderSemantic example that shows the instance tree of
+	the EBNF file <i>syntax/builder/examples/SyntaxBuilder.syntax</i<.
+*/
+
+public class TreeBuilderExample
+{
+	/** Test output of SyntaxBuilder syntax tree. */
+	public static void main(String [] args)	{
+		try	{
+			Reader input = new InputStreamReader(Parser.class.getResourceAsStream("syntax/builder/examples/SyntaxBuilder.syntax"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *input*. Currently, there are execution paths that do not contain closure statements, for example, when SyntaxSeparation constructor throws an exception. Ensure *input* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).